### ISSUE TITLE
`col.str.contains` does not respect case insensitive regex object

### ISSUE THREAD

**Issue by kymckay** [pola-rs/nodejs-polars#82]
#### Have you tried latest version of polars?

- yes

#### What version of polars are you using?

`0.8.0`

#### What operating system are you using polars on?

Debian

#### What node version are you using

v18.13.0

#### Describe your bug.

Using the `str.contains` function of a column doesn't respect the case insensitive flag of a provided regex object.

#### What are the steps to reproduce the behavior?

Use str.contains on a column, providing a regex object created with the `"i"` flag for case insensitivity. Test it on a sample with different casing than the original regex pattern.

**Example**

```js
import pl from "nodejs-polars"

let df = pl.DataFrame({
    "text": ["foo", "FOO", "FoO"],
})

const regex = new RegExp("foo", "i")

df = df.withColumn(pl.col("text").str.contains(regex).alias("result"))

console.log(df.toString())
```

#### What is the actual behavior?

The contains function does not match all case variations.

```
shape: (3, 2)
┌──────┬────────┐
│ text ┆ result │
│ ---  ┆ ---    │
│ str  ┆ bool   │
╞══════╪════════╡
│ foo  ┆ true   │
│ FOO  ┆ false  │
│ FoO  ┆ false  │
└──────┴────────┘
```

#### What is the expected behavior?

The contains function should match all variations. Probably by injecting the appropriate `(?i)` and `(?-i)` flags for polars to interpret.

```
shape: (3, 2)
┌──────┬────────┐
│ text ┆ result │
│ ---  ┆ ---    │
│ str  ┆ bool   │
╞══════╪════════╡
│ foo  ┆ true   │
│ FOO  ┆ true   │
│ FoO  ┆ true   │
└──────┴────────┘
```

**Comment by kymckay** [comment]
For what it's worth this is easily worked around using a string and the `(?i)`, `(?-i)` flags as needed. I'm just raising for posterity.

**Comment by Bidek56** [comment]
This is not an easy one b/c we need to translate Javascript Regex syntax to Rust regex.

**Comment by universalmind303** [comment]
closed via #92
